src/main/java/it/pagopa/transactions/utils/NodoOperations.java

```diff
@@ -163,7 +163,8 @@ private Mono<PaymentRequestInfo> nodoActivationForNM3PaymentRequest(
                                                         transfer.getTransferCategory()
                                                 )
                                         ).toList(),
-                                isAllCCP(response, allCCPOnTransferIbanEnabled)
+                                isAllCCP(response, allCCPOnTransferIbanEnabled),
+                                response.getCreditorReferenceId()
```

Comment:
I would add a Nodo semantic response validation here. For WISP redirect transactions we expect that `creditorReferenceId` would be always valued, a null value for those cases will lead to an error since we would not have a value to be shown to the client (checkout) and to be sent into the email body.

So for wisp redirect transaction this value should be present and not null here (it's nullable into specs) and for null response value I would throw an `InvalidNodoResponseException` here

Reply:
Solved in https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/558/commits/b55bcbb47b9d4b75ec1e2a97af89d1aeab4eed45 by wrapping nodo activate payment